static/js/components/ShiftPage.jsx

```diff
@@ -0,0 +1,129 @@
+import Grid from "@material-ui/core/Grid";
+import { makeStyles } from "@material-ui/core/styles";
+import PropTypes from "prop-types";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import NewShift from "./NewShift";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: "100%",
+    maxWidth: "22.5rem",
+    backgroundColor: theme.palette.background.paper,
+  },
+  paperContent: {
+    padding: "1rem",
+  },
+}));
+
+const textStyles = makeStyles(() => ({
+  primary: {
+    fontWeight: "bold",
+    fontSize: "110%",
+  },
+}));
+
+export function shiftTitle(shift) {
+  if (!shift?.group.name) {
+    return (
+      <div>
+        <CircularProgress color="secondary" />
+      </div>
+    );
+  }
+
+  let result = `${shift?.group.name}`;
+  if (shift?.name) {
+    result += `: ${shift?.name}`;
+  }
+
+  return result;
+}
+
+export function shiftInfo(shift) {
+  if (!shift?.group.name) {
+    return (
+      <div>
+        <CircularProgress color="secondary" />
+      </div>
+    );
+  }
+
+  const array = [
+    ...(shift?.start_date ? [`Start Date: ${shift.start_date}`] : []),
+    ...(shift?.end_date ? [`End Date: ${shift.end_date}`] : []),
+  ];
+
+  // eslint-disable-next-line prefer-template
+  const result = "( " + array.join(" / ") + " )";
+
+  return result;
+}
+
+const ShiftList = ({ shifts }) => {
+  const classes = useStyles();
+  const textClasses = textStyles();
+
+  return (
+    <div className={classes.root}>
+      <List component="nav">
+        {shifts?.map((shift) => (
+          <ListItem button key={shift.id}>
+            <ListItemText
+              primary={shiftTitle(shift)}
+              secondary={shiftInfo(shift)}
+              classes={textClasses}
+            />
+          </ListItem>
+        ))}
+      </List>
+    </div>
+  );
+};
+
+const ShiftPage = () => {
+  const classes = useStyles();
+  const { shiftList } = useSelector((state) => state.shifts);
+  const currentUser = useSelector((state) => state.profile);
+
+  if (!shiftList) {
+    return <CircularProgress />;
+  }
+
+  return (
+    <Grid container spacing={3}>
+      <Grid item md={6} sm={12}>
+        <Paper elevation={1}>
+          <div className={classes.paperContent}>
+            <Typography variant="h6">List of Shifts</Typography>
+            <ShiftList shifts={shiftList} />
+          </div>
+        </Paper>
+      </Grid>
+      {(currentUser.permissions?.includes("System admin") ||
+        currentUser.permissions?.includes("Manage groups") ||
+        currentUser.permissions?.includes("Manage shifts")) && (
+        <Grid item md={6} sm={12}>
+          <Paper>
+            <div className={classes.paperContent}>
+              <Typography variant="h6">Add a New Shift</Typography>
+              <NewShift />
+            </div>
+          </Paper>
+        </Grid>
+      )}
+    </Grid>
+  );
+};
+
+ShiftList.propTypes = {
+  shifts: PropTypes.arrayOf(PropTypes.any).isRequired,
+};
+
```

Comment:
Should have a way to delete shifts, particularly the ones that a user added themselves.

Reply:
You mean on the front end I guess? Because obviously can do that through the API.